src/main/java/io/cryostat/core/net/JFRConnection.java

```diff
@@ -146,6 +156,35 @@ public synchronized int getPort() {
         }
     }
 
+    public synchronized String getJvmId() throws Exception {
+        if (!isConnected()) {
+            connect();
+        }
+        ObjectName runtimeBean = new ObjectName("java.lang:type=Runtime");
+        List<String> attrNames =
+                new ArrayList<>(
+                        Arrays.asList(
+                                "ClassPath",
+                                "Name",
+                                "InputArguments",
+                                "LibraryPath",
+                                "VmVendor",
+                                "VmVersion",
+                                "StartTime"));
+
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream(512);
+                DataOutputStream dos = new DataOutputStream(baos)) {
+            for (String attr : attrNames) {
+                dos.writeUTF(
+                        this.rjmxConnection
+                                .getAttributeValue(new MRI(Type.ATTRIBUTE, runtimeBean, attr))
+                                .toString());
+            }
```

Comment:
I just had a thought here - might it make sense to include the machine's IP address in the stream before hashing it? I don't think we can retrieve that in the same uniform way through the Runtime bean, but it can still be appended after all those attributes are written in.

In the case of long-lived deployments on bare metal the IP address is very likely to be stable long term, not changing during the target JVM lifecycle anyway. In the case of deployments on a container platform like OpenShift then each replica of a target service will have its own unique IP address since they'd be sitting behind a load balancer, so this would eliminate the narrow timing window possibility where multiple replicas come up at the exact same `StartTime` (since all other attributes would also be the same for replicas of a service). If the deployment is scaled down or a replica crahses then we should notice that from the target discovery mechanism and can use that to clear any cached computed IDs, so even if a replacement replica comes online later (restart or scale back up) and the same IP happens to be re-assigned, we can treat it as a "fresh" target since the caches have already been cleared.

Reply:
Sounds good. I'll add that in

Reply:
I guess it's possible, especially on bare metal deployments, that there are actually multiple IP addresses for multiple network interfaces. I guess all of those should be put into the hash algo.

Reply:
Sorry, where are the machine IPs located? None of the mbeans seem to show networking info.

Reply:
Shoot, sorry for the wild goose chase. I took a look and couldn't seem to find any way to access it either. I thought at the least we could grab it from the open JMX connection itself, but even that doesn't look like it'll work.

Forget about it for now. I'll keep thinking if there are any ways to do this, or other things we can add to the hash input.

Reply:
Okay no worries